messaging/src/main/java/org/axonframework/configuration/ComponentRegistry.java

```diff
@@ -180,6 +180,65 @@ default boolean hasComponent(@Nonnull Class<?> type) {
     boolean hasComponent(@Nonnull Class<?> type,
                          @Nonnull String name);
 
+    /**
+     * Registers a {@link Component} only <b>if</b> there is none yet for the given {@code type}.
+     * <p>
+     * The given {@code builder} function gets the {@link Configuration configuration} as input, and is expected to
+     * provide the component as output. The component will be registered under an {@link Identifier} based on the given
+     * {@code type}.
+     *
+     * @param type    The declared type of the component to build, typically an interface.
+     * @param builder The builder building the component.
+     * @param <C>     The type of component the {@code builder} builds.
+     * @return The current instance of the {@code Configurer} for a fluent API.
+     * @throws ComponentOverrideException If the override policy is set to
```

Comment:
This `throws` does not make sense right? As it doesn't register if it exists, so it doesn't override. 

It's also on the other added methods

Reply:
You are entirely right...copy-paste error here. I'll remove these right away.